src/main/scala/is/hail/expr/ir/AggOp.scala

```diff
@@ -134,6 +135,50 @@ object AggOp {
       case _ => CodeAggregator[RegionValueTakeAnnotationAggregator](in, TArray(in), constrArgTypes = Array(classOf[Int], classOf[Type]))
     }
 
+    case (TakeBy(), in, constArgs@Seq(_: TInt32), None, Seq(key)) => (in, key) match {
+      case (_: TBoolean, _: TBoolean) => CodeAggregator[RegionValueTakeByBooleanBooleanAggregator] (in, TArray(in), constrArgTypes = Array(classOf[Int], classOf[Type], classOf[Type]), seqOpArgTypes = Array(classOf[Boolean]))
+      case (_: TBoolean, _: TInt32 | _: TCall) => CodeAggregator[RegionValueTakeByBooleanIntAggregator] (in, TArray(in), constrArgTypes = Array(classOf[Int], classOf[Type], classOf[Type]), seqOpArgTypes = Array(classOf[Int]))
```

Comment:
This is just formatting, but this is pretty hard to read. What about something like this?
```Scala
    case (TakeBy(), in, constArgs@Seq(_: TInt32), None, Seq(key)) => {
      def mkAgg(t: Class[_]) =
        CodeAggregator[RegionValueTakeByBooleanBooleanAggregator](in, TArray(in),
          constrArgTypes = Array(classOf[Int], classOf[Type], classOf[Type]),
          seqOpArgTypes = Array(t))

      (in, key) match {
        case (_: TBoolean, _: TBoolean) => mkAgg(classOf[Boolean])
        case (_: TBoolean, _: TInt32 | _: TCall) => mkAgg(classOf[Int])
        // ...
      }
    }
```